lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1439,9 +1439,18 @@ static LogicalResult foldSingleSetConnect(ConnectOp op,
   if (srcValueOp && srcValueOp != &declBlock->front())
     srcValueOp->moveBefore(&declBlock->front());
 
+  // Replace with constant zero
```

Comment:
This is a folder, does it trigger during parsing?

Reply:
Looks like its only called from the canonicalizer, I was very scared for a second.  Maybe we should change this function name to `canonicalizeSingleSetConnect`.

Reply:
Agree about the name change. This confused me and I had the same ❗ reaction. I can change this is an NFC commit.

Reply:
Landed separately in: 635786e1cb8ce3b2c4d4a75be45b083944d2f471.